source/nodejs/adaptivecards/src/carousel.ts

```diff
@@ -0,0 +1,296 @@
+            // https://stackoverflow.com/questions/36247140/why-doesnt-flex-item-shrink-past-content-size
+            swiperWrapper.style.minHeight = '-webkit-min-content';
+        }
+
+        // switch (this.getEffectiveVerticalContentAlignment()) {
+        //     case VerticalAlignment.Center:
+        //         swiperWrapper.style.justifyContent = "center";
+        //         break;
+        //     case VerticalAlignment.Bottom:
+        //         swiperWrapper.style.justifyContent = "flex-end";
+        //         break;
+        //     default:
+        //         swiperWrapper.style.justifyContent = "flex-start";
+        //         break;
+        // }
+
+        for (const page of this._pages) {
+            const renderedItem = this.isElementAllowed(page) ? page.render() : undefined;
+            if (renderedItem) {
+                Utils.appendChild(swiperWrapper, renderedItem);
+                this._renderedPages.push(page);
+            }
+        }
+
+        swiperContainer.appendChild(swiperWrapper as HTMLElement);
+
+        const prevElementDiv: HTMLElement = document.createElement("button");
+        prevElementDiv.classList.add("swiper-button-prev");
+        swiperContainer.appendChild(prevElementDiv);
+
+        const nextElementDiv: HTMLElement = document.createElement("button");
+        nextElementDiv.classList.add("swiper-button-next");
+        swiperContainer.appendChild(nextElementDiv);
+
+        const pagination: HTMLElement = document.createElement("div");
+        pagination.classList.add("swiper-pagination");
+        swiperContainer.appendChild(pagination);
+
+        this.initializeSwiper(swiperContainer, nextElementDiv, prevElementDiv, pagination);
+
+        cardLevelContainer.appendChild(swiperContainer);
+        return this._renderedPages.length > 0 ? cardLevelContainer : undefined;
+    }
+
+    private initializeSwiper(swiperContainer: HTMLElement, nextElement: HTMLElement, prevElement: HTMLElement, paginationElement: HTMLElement): void {
+        const swiperOptions: SwiperOptions = {
+            loop: true,
+            pagination: {
+                el: paginationElement
+            },
+            navigation: {
+                prevEl: prevElement,
+                nextEl: nextElement
+            },
+            a11y: {
+                enabled: true
+            },
+            keyboard: {
+                enabled: true,
+                onlyInViewport: true
+            }
+        };
+
+        if (this.timer && !this.isDesignMode()) {
+            swiperOptions.autoplay = { delay: this.timer };
+        }
+
+        new Swiper(swiperContainer, swiperOptions);
+    }
+
+    private createCarouselPageInstance(source: any, context: SerializationContext): CarouselPage | undefined {
+        return context.parseCardObject<CarouselPage>(
+            this,
+            source,
+            CarouselPage.unsupportedElements,
```

Comment:
>CarouselPage.unsupportedElements [](http://example.com/codeflow?start=12&length=32)

is this array propagated? for example, in ColumnSets, where the parsing of Columns is done differently (the same issue may happen on Tables)

Reply:
I'm currently reworking this stuff (it was only half-implemented here) -- I'll unmark this PR as draft once it's good to go :)